packages/core/src/zipHardhat.ts

```diff
@@ -0,0 +1,160 @@
+import JSZip from "jszip";
+import type { GenericOptions } from "./build-generic";
+import type { Contract } from "./contract";
+import type { ERC1155Options } from "./erc1155";
+import { printContract } from "./print";
+
+const hardhatConfig = (upgradeable: boolean) => `\
+import { HardhatUserConfig } from "hardhat/config";
+import "@nomicfoundation/hardhat-toolbox";${upgradeable ? `
+import "@openzeppelin/hardhat-upgrades";` : ''}
+
+const config: HardhatUserConfig = {
+  solidity: {
+    version: "0.8.4",
+    settings: {
+      optimizer: {
+        enabled: true,
+      },
+    },
+  }
+};
+
+export default config;
+`;
+
+const tsConfig = `\
+{
+  "compilerOptions": {
+    "target": "es2020",
+    "module": "commonjs",
+    "esModuleInterop": true,
+    "forceConsistentCasingInFileNames": true,
+    "strict": true,
+    "skipLibCheck": true
+  }
+}
+`
+
+const gitIgnore = `\
+node_modules
+.env
+coverage
+coverage.json
+typechain
+typechain-types
+
+#Hardhat files
+cache
+artifacts
+`
+
+const test = (c: Contract, opts?: GenericOptions) => {
+
+let buf = `\
+import { expect } from "chai";
+import { ethers${c.upgradeable ? ', upgrades' : ''} } from "hardhat";
+
+describe("${c.name}", function () {
+  it("Test contract", async function () {
+    const ContractFactory = await ethers.getContractFactory("${c.name}");
+
+    const instance = await ${c.upgradeable ? 'upgrades.deployProxy(ContractFactory);' : 'ContractFactory.deploy();' }
+    await instance.deployed();
+
+`;
+
+if (opts !== undefined) {
+  switch (opts.kind) {
+    case 'ERC20':
+    case 'ERC721':
+      buf += `    expect(await instance.name()).to.equal("${opts.name}");`
+      break;
+
+    case 'ERC1155':
+      buf += `    expect(await instance.uri(0)).to.equal("${(opts as ERC1155Options).uri}");`
```

Comment:
Cast not necessary, it gets narrowed automatically (I checked).

```suggestion
      buf += `    expect(await instance.uri(0)).to.equal("${opts.uri}");`
```